File: shared/config/tables.py
# テーブル名定数
# SQLインジェクション対策のホワイトリスト
# 全てのテーブル名はここで一元管理する

from typing import Set

# 物件関連テーブル（CRUD操作対象）
PROPERTY_TABLES: Set[str] = {
    "properties",
    "building_info",
    "land_info",
    "property_images",
    "property_locations",
    "property_registries",
    "amenities",
}

# メタデータテーブル（管理画面で参照）
METADATA_TABLES: Set[str] = {
    "column_labels",
    "master_categories",
    "master_options",
}

# マスタテーブル（参照のみ）
MASTER_TABLES: Set[str] = {
    "m_facilities",
    "m_stations",
    "m_postal_codes",
    "m_roles",
}

# 全許可テーブル（metadata API用）
ALL_ALLOWED_TABLES: Set[str] = PROPERTY_TABLES | METADATA_TABLES | MASTER_TABLES

# GenericCRUD用（CRUD操作を許可するテーブル）
CRUD_ALLOWED_TABLES: Set[str] = PROPERTY_TABLES

File: rea-admin/src/constants/tables.ts
// テーブル名定数
// バックエンドのshared/config/tables.pyと同期すること

// 物件関連テーブル（フォーム表示対象）
export const PROPERTY_FORM_TABLES = [
  'properties',
  'building_info',
  'land_info',
  'amenities',
  'property_images',
] as const;

// 物件関連全テーブル（CRUD対象）
export const PROPERTY_TABLES = [
  ...PROPERTY_FORM_TABLES,
  'property_locations',
  'property_registries',
] as const;

// 型定義
export type PropertyFormTable = typeof PROPERTY_FORM_TABLES[number];
export type PropertyTable = typeof PROPERTY_TABLES[number];
